crates/uv-cli/src/lib.rs

```diff
@@ -427,6 +427,10 @@ pub enum SelfCommand {
 pub struct SelfUpdateArgs {
     /// Update to the specified version. If not provided, uv will update to the latest version.
     pub target_version: Option<String>,
+
+    /// Specify a github token for authentication during the update process.
+    #[arg(long, env = "UV_GITHUB_TOKEN")]
```

Comment:
I think we could probably just respect `GITHUB_TOKEN` here since it's standardized

Reply:
We'll also need to update `docs/configuration/environment.md`

Reply:
axoupdate [recommends against that](https://github.com/axodotdev/axoupdater?tab=readme-ov-file#github-actions-and-rate-limits-in-ci).

> We also recommend picking an environment variable name that's specific to your application; it's not uncommon for users to have stale or expired `GITHUB_TOKEN` tokens in their environment, and using that name may cause your app to behave unexpectedly.

Reply:
Hm, interesting. Thanks for the reference. Let me see if anyone else has an opinion, but it's always easier to start with the scoped variable as you have here.

cc @charliermarsh

Reply:
An extra bonus is should we show friendly error message to remind people to set github token when rate limit is reached?

Reply:
> An extra bonus is should we show friendly error message to remind people of setting github token when rate limit is reached?

* * *

However, from the HTTP status code 403, we can only conclude that it might be rate limiting or an invalid token.

Reply:
I think if we encounter a 403 and weren't given a token we can show a rate limit hint, that seems really nice.